src/components/editor/DraftSpec.tsx

```diff
@@ -57,28 +54,18 @@ function DraftSpecEditor() {
     }, [draftSpecs, setSpecs]);
 
     useEffect(() => {
-        setDraftSpec(
-            draftSpecs[currentCatalog] ? draftSpecs[currentCatalog] : null
-        );
-    }, [currentCatalog, draftSpecs, setSpecs]);
+        if (currentCatalog) {
+            setDraftSpec(currentCatalog);
+        }
```

Comment:
We just store the current catalog now and not the index. So we don't want to set the draft spec unless there is actually a catalog to display.